library/L2_HAL/displays/lcd/st7066u.hpp

```diff
@@ -96,69 +90,51 @@ class St7066u
-                    DisplayMode display_mode,
-                    FontStyle font_style,
-                    const ControlPins_t & pins)
+  /// @param font_style Character font style
+  /// @param register_select_pin Pin that outputs whether the instruction or
+  ///                            data register is selected.
+  /// @param read_write_pin Pin that outputs whether the operation is
+  ///                       read/write.
+  /// @param enable_pin Enable pin used for triggering read/write operations.
+  /// @param data_bus 4-bit or 8-bit GPIO parallel bus depending on the
+  ///                 specified bus_mode.
+  explicit constexpr St7066u(BusMode bus_mode,
+                             DisplayMode display_mode,
+                             FontStyle font_style,
+                             sjsu::Gpio & register_select_pin,
+                             sjsu::Gpio & read_write_pin,
+                             sjsu::Gpio & enable_pin,
+                             ParallelBus & data_bus)
       : kBusMode(bus_mode),
         kDisplayMode(display_mode),
         kFontStyle(font_style),
-        kControlPins(pins)
+        kRegisterSelectPin(register_select_pin),
+        kReadWritePin(read_write_pin),
+        kEnablePin(enable_pin),
+        kDataBus(data_bus)
   {
   }
 
   /// Initialize the pins needed to communicate with the LCD screen
-  void Initialize()
+  void Initialize() const
   {
-    kControlPins.rs.SetDirection(sjsu::Gpio::Direction::kOutput);
-    kControlPins.rw.SetDirection(sjsu::Gpio::Direction::kOutput);
-    kControlPins.e.SetDirection(sjsu::Gpio::Direction::kOutput);
-    kControlPins.e.Set(sjsu::Gpio::kHigh);
-    kControlPins.d7.SetDirection(sjsu::Gpio::Direction::kOutput);
-    kControlPins.d6.SetDirection(sjsu::Gpio::Direction::kOutput);
-    kControlPins.d5.SetDirection(sjsu::Gpio::Direction::kOutput);
-    kControlPins.d4.SetDirection(sjsu::Gpio::Direction::kOutput);
-
-    if (kBusMode == BusMode::kEightBit)
-    {
-      kControlPins.d3.SetDirection(sjsu::Gpio::Direction::kOutput);
-      kControlPins.d2.SetDirection(sjsu::Gpio::Direction::kOutput);
-      kControlPins.d1.SetDirection(sjsu::Gpio::Direction::kOutput);
-      kControlPins.d0.SetDirection(sjsu::Gpio::Direction::kOutput);
-    }
+    kRegisterSelectPin.SetDirection(sjsu::Gpio::Direction::kOutput);
+    kReadWritePin.SetDirection(sjsu::Gpio::Direction::kOutput);
+    kEnablePin.SetDirection(sjsu::Gpio::Direction::kOutput);
+    kEnablePin.Set(sjsu::Gpio::kHigh);
+    kDataBus.Initialize();
+    kDataBus.SetDirection(sjsu::Gpio::Direction::kOutput);
 
-    WriteCommand(static_cast<uint8_t>(Command::kDefaultDisplayConfiguration) |
-                 static_cast<uint8_t>(kBusMode) |
-                 static_cast<uint8_t>(kDisplayMode) |
-                 static_cast<uint8_t>(kFontStyle));
+    WriteCommand(Value(Command::kDefaultDisplayConfiguration) |
+                 Value(kBusMode) | Value(kDisplayMode) | Value(kFontStyle));
```

Comment:
```suggestion
    WriteCommand(Value(Command::kDefaultDisplayConfiguration) |
                 Value(kBusMode) | Value(kDisplayMode) | Value(kFontStyle));
```

See `enum.hpp` and look for the `SJ2_ENABLE_BITMASK_OPERATORS` so that you can OR these together without having to use Value so often. Also

If you use that, then you won't even need to use the `Value()` method for the enums becaue you can have `WriteCommand` take a composed `Command` enum object.

Reply:
Unfortunately this doesn't seem to work for multiple different enums.

Reply:
Ah that makes sense. If they are not the same enum class then my eable bitmask operators does not work.